app/src/main/java/com/team7/cmput301/android/theirisproject/task/CacheBodyPhotoTask.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Team 7, CMPUT301, University of Alberta - All Rights Reserved. You may use, distribute, or modify this code under terms and conditions of the Code of Students Behavior at University of Alberta
+ */
+
+package com.team7.cmput301.android.theirisproject.task;
+
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import com.team7.cmput301.android.theirisproject.ImageConverter;
+import com.team7.cmput301.android.theirisproject.model.BodyPhoto;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * CacheBodyPhotoTask will run in the background to save body photos
+ * in the apps internal storage for quicker loading of images on
+ * BodyPhotoListActivity
+ *
+ * @see com.team7.cmput301.android.theirisproject.activity.BodyPhotoListActivity
+ * @author itstc
+ * */
+public class CacheBodyPhotoTask extends AsyncTask<BodyPhoto, Void, Void> {
+
+    private Context context;
+    public CacheBodyPhotoTask(Context context) {
```

Comment:
The application context is set in IrisProjectApplication (and it's the same throughout the entire app's life-cycle, apparently), so you can make a getter and use that here if you want, to avoid relaying the context from activity to controller to task.